packages/kotti-ui/source/kotti-field-inline-edit/KtFieldInlineEdit.vue

```diff
@@ -0,0 +1,339 @@
+<template>
+	<KtField
+		v-bind="{ field }"
+		ref="fieldRef"
+		:class="rootClasses"
+		:getEmptyValue="() => null"
+		:helpTextSlot="$slots.helpText"
+	>
+		<EditIcon v-if="!isEditing" ref="editIconRef" />
+		<textarea
+			v-if="isMultiline"
+			ref="inputRef"
+			v-bind="inputProps"
+			@input="onEdit"
+			@keydown.enter="onEnterMultiline"
+			@keyup.esc="onCancel"
+		/>
+		<input
+			v-else
+			ref="inputRef"
+			v-bind="inputProps"
+			@input="onEdit"
+			@keydown.enter="onConfirm"
+			@keyup.esc="onCancel"
+		/>
+		<ConfirmButton
+			v-if="isEditing"
+			ref="confirmButtonRef"
+			slot="container-right"
+			v-bind="{ isEditing }"
+			:tabIndex="tabIndex"
+			@confirm="onConfirm"
+		/>
+	</KtField>
+</template>
+
+<script lang="ts">
+import { Yoco } from '@3yourmind/yoco'
+import {
+	computed,
+	defineComponent,
+	onBeforeMount,
+	onUnmounted,
+	ref,
+	watch,
+} from '@vue/composition-api'
+
+import { useField, useForceUpdate } from '../kotti-field/hooks'
+import { useTranslationNamespace } from '../kotti-i18n/hooks'
+import { makeProps } from '../make-props'
+
+import ConfirmButton from './components/ConfirmButton.vue'
+import EditIcon from './components/EditIcon.vue'
+import { KOTTI_FIELD_INLINE_EDIT_SUPPORTS } from './constants'
+import { useAdjustHeight } from './hooks'
+import { KottiFieldInlineEdit, _ComponentRef } from './types'
+import { blurField, isFieldTriggered, isInFocus } from './utils'
+
+export default defineComponent<KottiFieldInlineEdit.PropsInternal>({
+	name: 'KtFieldInlineEdit',
+	components: {
+		ConfirmButton,
+		EditIcon,
+	},
+	props: makeProps(KottiFieldInlineEdit.propsSchema),
+	// eslint-disable-next-line sonarjs/cognitive-complexity
+	setup(props, { emit }) {
+		onBeforeMount(() => {
+			window.addEventListener('click', onClick)
+			window.addEventListener('focus', onFocusChange, true)
+		})
+
+		onUnmounted(() => {
+			window.removeEventListener('click', onClick)
+			window.removeEventListener('focus', onFocusChange)
+		})
+
+		const { forceUpdate, forceUpdateKey } = useForceUpdate()
+		const translations = useTranslationNamespace('KtFieldInlineEdit')
+
+		const preEditingValue = ref<KottiFieldInlineEdit.Value>(props.value)
+
+		const confirmButtonRef = ref<_ComponentRef>(null)
+		const fieldRef = ref<_ComponentRef>(null)
+		const inputRef = ref<HTMLInputElement | HTMLTextAreaElement | null>(null)
+		const editIconRef = ref<_ComponentRef>(null)
```

Comment:
`confirmButtonRef` and `editIconRef` seem not to be used anywhere. i only see 3 instances of each (here, in the return, and in the template)